src/components/PivotTable/PivotTableGrid/model.ts

```diff
@@ -0,0 +1,121 @@
+import { KeyMap } from '../model'
+import { PivotTableCellProps } from '../PivotTableCell/PivotTableCell'
+
+export interface PivotTableProps<T extends object> {
+  /**
+   * Table keys mapped to their display names, value formatters and value ordenators
+   */
+  keysMapping: KeyMap<T>
+  /**
+   * A tree with the count of each occurence of a value relative to parent keys.
+   * If the complementaryTree is not null, it is used to build the columns of the table
+   */
+  defaultTree: PivotTableTreeNode<T>
+  /**
+   * A tree with the count of each occurrence of a value relative to parent keys. Used to build the rows of the table
+   */
+  complementaryTree?: PivotTableTreeNode<T>
+  /**
+   * Ordered list of keys to be displayed as rows
+   */
+  rowKeys?: Array<keyof T>
+  /**
+   * List of keys to be displayed as columns
+   */
+  columnKeys?: Array<keyof T>
+}
+
+export interface PivotTableTreeNodeValues {
+  /**
+   * Category of data that the children of this node belong to
+   */
+  nodeKey?: string
+  /**
+   * Count of values belonging to this node of the tree
+   */
+  nodeValue?: number
+  /**
+   * Biggest leaf value under this node. Used to color the table cells
+   */
+  maxLeafValue?: number
+}
+
+export type PivotTableTreeNodeChildren<T extends any> = Record<string | number | symbol, T[] & PivotTableTreeNodeValues>
+export type PivotTableTreeNode<T extends any> = PivotTableTreeNodeChildren<T> & PivotTableTreeNodeValues
+
+export interface TreeMeta<T> {
+  isEmpty: boolean
+  numberKeys: string[]
+  keyValues: Map<keyof T, Array<string>>
+}
+
+export interface CellInitialPosition {
+  parentInitialPosition?: CellInitialPosition
+  auxInitialPosition?: CellInitialPosition
+  cellSpan?: SpanValue
+}
+
+export interface CellData<T> {
```

Comment:
`Data` is kinda redundant in the name of an interface :sweat_smile: I think `PivotTableCellModel`is a better name.